File: app/Models/Tenant/ServiceProvider.php
<?php

namespace App\Models\Tenant;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\SoftDeletes;

class ServiceProvider extends Model
{
    use SoftDeletes;

    protected $connection = 'society';

    protected $fillable = [
        'name',
        'service_type',
        'phone',
        'notes',
        'sort_order',
    ];
}

File: mobile/lib/features/service_providers/data/service_provider.dart
class ServiceProvider {
  ServiceProvider({
    required this.id,
    required this.name,
    required this.serviceType,
    required this.phone,
    this.notes,
  });

  factory ServiceProvider.fromJson(Map<String, dynamic> json) {
    return ServiceProvider(
      id: json['id'] as int,
      name: json['name'] as String,
      serviceType: json['service_type'] as String,
      phone: json['phone'] as String,
      notes: json['notes'] as String?,
    );
  }

  final int id;
  final String name;
  final String serviceType;
  final String phone;
  final String? notes;
}
